README.md

```diff
@@ -87,16 +87,67 @@ curl -i http://localhost:8000/avatars/head:beluga/tail:fish/color:%2331688e/500x
 
 #### `/games/{game id}/gif`
 
+**DEPRECATED:** use `/games/{game id}/{width}x{height}.gif` instead.
+
 Exports the game as an animated gif.
 
 #### `/games/{game id}/frames/{frame number}/gif`
 
+**DEPRECATED:** use `/games/{game id}/frames/{frame number}/{width}x{height}.gif` instead.
+
 Exports a specific frame as a gif (no animation).
 
 #### `/games/{game id}/frames/{frame number}/ascii`
 
+**DEPRECATED:** use `/games/{game id}/frames/{frame number}.txt` instead.
+
+Exports a specific frame as an ASCII string.
+
+#### `/games/{game id}/{width}x{height}.gif`
+
+Exports the game as an animated gif sized `width` pixels wide and `height` pixels high.
+
+See [GIF size validation](#Choose-a-GIF-size) for details about how to choose a valid GIF resolution
+
+#### `/games/{game id}/frames/{frame number}/{width}x{height}.gif`
+
+Exports the game as an animated gif sized `width` pixels wide and `height` pixels high.
+
+See [GIF size validation](#Choose-a-GIF-size) for details about how to choose a valid GIF resolution
+
+#### `/games/{game id}/frames/{frame number}.txt`
+
 Exports a specific frame as an ASCII string.
 
+### Choose a GIF size
+
+GIF sizes are restricted to a limited set of options based on the game board being exported. Additionally, there is an upper-limit of a maximum resolution of `504x504` (`254016` pixels) which supersedes the calculation of available options.
+
+
+At the time of this writing, the options are:
+
+- 10 pixels per board square (+ 4 pixels for border)
+- 20 pixels per board square (+ 4 pixels for border)
+- 30 pixels per board square (+ 4 pixels for border)
+- 40 pixels per board square (+ 4 pixels for border)
+
+These options may change. If you need an up-to-date list, look at `allowedPixelsPerSquare` in [handlers.go](./http/handlers.go) for an authoritative list of allowed resolutions.
+
+Using the above, you can determine the available GIF dimensions that you can request.
+
+Examples:
+- allowed sizes allowed for **11x11** board are: 
+  - **114x114** (10 pixels per board square)
+  - **224x224** (20 pixels per board square)
+  - **334x334** (30 pixels per board square)
+  - **444x444** (40 pixels per board square)
+- allowed sizes allowed for **19x19** board are: 
+  - **194x194** (10 pixels per board square)
+  - **384x384** (20 pixels per board square)
+  - **574x574** (30 pixels per board square) (**Disallowed** because it exceeds `504x504`)
+  - **764x764** (40 pixels per board square) (**Disallowed** because it exceeds `504x504`)
+- etc...
+
```

Comment:
Since there's no common allowed board size, is the idea here that Play will need to do a calculation to generate the URL for a preset size in order to link the GIF for a game? e.g. we'd decide we want the "View GIF" button to link to the "30 pixels per board square" resolution, and generate the URL accordingly on the fly?

Reply:
Good point. Do you think it would be better to have the parameters be the resolution per-square instead?

Reply:
Good point. Do you think it would be better to have the dimension parameters be the resolution per-square instead? I feel like that would be better so Play wouldn't have to do those calculations.

I originally started with allowing any size. Then after some discussion, it seemed that limited options were a good idea, but there wasn't a good way to do this with static sizes, so I came up with this.

Reply:
I'm not sure which is better, TBH. There's something about having the actual image size reflected in the URL that seems convenient. I don't think it's a big deal for play to include that calculation. Might be worth asking in #dev in case there's a use case I've missed here.

Reply:
I brought this up with the team this morning, and the consensus was that we haven't fully worked out the use cases for different GIF sizes, so we can just ship as-is for now. We may end up revisiting the endpoint design with something like small/medium/large mapping to fixed pixels-per-square values, which I think won't be incompatible with the routing approach used here?

Reply:
Sounds good Rob. Yeah I think the routing is quite flexible. It should be easy enough to adapt this to other parameters in the future.